geoproxy/main.go

```diff
@@ -95,42 +113,117 @@ func parseRequestBody(request *http.Request) requestPayloadStruct {
 	return requestPayload
 }
 
-// Log the typeform payload and redirect url
-func logRequestPayload(requestionPayload requestPayloadStruct, proxyUrl string) {
-	log.Printf("lat: %s, lng: %s, proxy_url: %s\n", requestionPayload.Lat, requestionPayload.Lng, proxyUrl)
+// Parse the submit requests body
+func parseSubmitRequestBody(request *http.Request) submitRequestPayloadStruct {
+	decoder := requestBodyDecoder(request)
+
+	var requestPayload submitRequestPayloadStruct
+	err := decoder.Decode(&requestPayload)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return requestPayload
 }
 
-// Get the url for a given lat and long pair
-func getProxyUrl(latRaw string, lngRaw string) string {
-	lat1, err1 := strconv.ParseFloat(latRaw, 32)
-	lng1, err2 := strconv.ParseFloat(lngRaw, 32)
-	lat := float32(lat1)
-	lng := float32(lng1)
+// Log the view typeform payload and redirect url
+func logViewRequestPayload(requestionPayload viewRequestPayloadStruct, proxyUrl string) {
+	log.Printf("latlng1: %s, latlng2: %s, proxy_url: %s\n", requestionPayload.LatLng1, requestionPayload.LatLng2, proxyUrl)
+}
 
-	default_condtion_url := os.Getenv("DEFAULT_CONDITION_URL")
+// Log the submit typeform payload and redirect url
+func logSubmitRequestPayload(requestionPayload submitRequestPayloadStruct, proxyUrl string) {
+	log.Printf("coordinate: %s, proxy_url: %s\n", requestionPayload.LatLng, proxyUrl)
+}
 
-	if err1 != nil || err2 != nil {
-		log.Printf("Error: Invalid lat-long pair.")
-		return default_condtion_url
+// Given string in form [0.0, 0.0], create and return coord struct
+func parseCoord(coord string) Coord {
+	splitCoord := strings.Split(coord, ",")
+	// We should have someething like {"[0.0", "0.0]"}
+	if len(splitCoord) != 2 {
+		// Returning error coord of 360, 360 for now
+		log.Printf("Error: Coordinate passed into json request should be of form [0.0, 0.0].")
+		return Coord{360, 360}
+	} else {
+		coord0 := []rune(splitCoord[0])
+		coord1 := []rune(splitCoord[1])
+
+		lat, errLat := strconv.ParseFloat(string(coord0[1:len(splitCoord[0])]), 64)
+		lng, errLng := strconv.ParseFloat(string(coord1[0:len(splitCoord[1])-1]), 64)
+
+		if errLat != nil || errLng != nil {
+			log.Printf("Error: Invalid lat-long coordinate.")
+			return Coord{360, 360}
+		}
+		return Coord{lat, lng}
 	}
+}
+
+func rangesOverlap(start1 float64, end1 float64, start2 float64, end2 float64) bool {
+	return math.Min(end1, end2) >= math.Max(start1, start2)
```

Comment:
I might be missing something but I think this doesn't work. What if we have:
start1=10, start2=5
end1 = 0, end2=4
Won't it say that they don't overlap?

Reply:
Oh wait end is always supposed be greater than start, correct? Bc in that case this should work

Reply:
Yep, this is assuming that end > start, but again we should check that when we handle invalid requests sent to us!

Reply:
I think we should deal with validation in a separate pr? I feel like this is fine for now